jetty-start/src/main/java/org/eclipse/jetty/start/StartArgs.java

```diff
@@ -1331,10 +1357,12 @@ public void parse(final String rawarg, String source)
         // Anything else with a "-" is considered a JVM argument
         if (arg.startsWith("-"))
         {
+            StartLog.debug("Unrecognized Arg: %s (from %s)", arg, source);
+
             // Only add non-duplicates
-            if (!jvmArgs.contains(arg))
+            if (!jvmArgSources.containsKey(arg))
             {
-                jvmArgs.add(arg);
+                jvmArgSources.put(arg, source);
```

Comment:
Use `putIfAbsent()` here.

Reply:
Disagree.
This arg can be overridden by a the same arg later on, which could have a different source.
The last source is what's used, so we replace each time to know the eventually used source.

Reply:
I don't understand? There is a `!containsKey()` followed by `put()` so it's equivalent to `putIfAbsent()`.

Reply:
You are correct that `putIfAbsent` is an appropriate replacement for what you reviewed.
I disagree that it should be used at all, the check `!containsKey()` shouldn't even be there too.
I pushed an update.